src/main/java/com/to/ordermanagementservice/service/impl/SimpleOrderService.java

```diff
@@ -46,20 +47,19 @@ private OrderDetails collectOrderDetails(Order order) {
         return orderDetails;
     }
     private List<OrderItemDetails> collectOrderItemDetails (Integer orderId){
-        List<OrderItem> orderItems = orderItemRepository.getOrderItemsByOrderId(orderId);
-        List<OrderItemDetails> result = new ArrayList<>();
-        for (OrderItem  orderItem: orderItems){
-            OrderItemDetails orderItemDetails = new OrderItemDetails();
-            orderItemDetails.setOrderId(orderItem.getOrderId());
-            orderItemDetails.setId(orderItem.getId());
-            orderItemDetails.setUpdateAt(orderItem.getUpdatedAt());
-            orderItemDetails.setCreatedAt(orderItem.getCreatedAt());
-            orderItemDetails.setPrice(orderItem.getPrice());
-            orderItemDetails.setQuantity(orderItem.getQuantity());
-            orderItemDetails.setProductId(orderItem.getProductId());
-            result.add(orderItemDetails);
-        }
-        return result;
+        return orderItemRepository.getOrderItemsByOrderId(orderId).stream()
+                .map(orderItem -> convertOrderItemToOrderItemDetails(orderItem)).toList();
```

Comment:
the good idea would be to keep each method call for streams in serapate lines. e.g:
```java
return collection.stream()
    .filter()
    .map()
    .sorted()
    .collect();
```

Reply:
understand